### PR TITLE
parallel calls for vrops targets

### PR DESCRIPTION
**Pull request by viennaa**
threading introduced for multiple runs to improve runtime.
shared timeout for threading
logging how long each target took to fetch its data

### CODE REVIEW

File: inventory.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def parse_params(logger):
                       action="store_true", dest="loopback")
     parser.add_option("-s", "--sleep", help="specifiy sleep time for inventory builder, default: 1800", action="store",
                       dest="sleep")
+    parser.add_option("-t", "--timeout", help="specifiy timeout for fetching data from vROps, default: 600", action="store",
+                      dest="timeout")
```
Comment (richardtief) [inline inventory.py:30]: Is there a difference to `--sleep`?

File: InventoryBuilder.py [hunk 2]
```diff
@@ -118,9 +119,35 @@ def query_inventory_permanent(self):
...
+                        if t[0] not in joined_threads:
+                            joined_threads.setdefault(t[1], round(time.time() - start_time))
+                if len(joined_threads.keys()) >= len(threads):
+                    break
+                time.sleep(1)
+                current_time = time.time()
+            else:
+                still_running = [t for t in threads if t[0].is_alive()]
+                for running_thread in still_running:
+                    logger.info(f"Timeout {timeout}s reached for fetching {running_thread[1]}")
```
Comment (richardtief) [inline InventoryBuilder.py:146]: Alright, so use the timeout parameter to close the thread if it gets stuck forever?

**Review by richardtief — APPROVED** [review]
Overdue implementation, well implemented. I have a few questions about the code.

File: inventory.py (hunk 1 above)
Comment (viennaa) [reply inventory.py:30]: it is. it defines how much time we give the inventory to fetch individual target vrops instances. If it is a bigger AZ, it takes longer for the inventory to fetch the data.
This is more like a failsafe, if something goes wrong in the thread, we kill it after $timeout seconds

**Pull request merged**